src/components/ToppingsSelect/index.jsx

```diff
@@ -3,14 +3,29 @@ import Topping from "../Topping";
 import './style.css';
 
 const ToppingsSelect = ({ toppings }) => {
+  const [selectedToppings, setSelectedToppings] = useState(toppings);
+  const [count, setCount] = useState([]);
```

Comment:
stejně jako "price", tak count je derivovaný stav a netřeba si ho ukládat do stavu (derivovaný stav = informaci kterou můžeme snadno odvodit z toho co už ve stavu máme). V tuhle chvíli to odvodíš jako
```
const count  = selectedToppings.filter(topping => topping.selected);
```
To co děláš ty je až pokročilá technika kdy to pole "selectedToppings" by mělo moc moc záznamů a pokaždé ho procházet by trvalo dlouho. Tak pak se to může takhle "zduplikova". Ale jako základ bych chtěl abys přemýšlela tak že cokoli si můžeš odvodit z toho co už máš, tak znovu neukládat (protože když zduplikuješ informace hrozí že se ti jejich verze rozjedou při nějakém update a budeš hledat nepříjemné bugy. Vždycky je lepší mít jeden autoritativní zdroj nějaké informace)